pkg/api/v1alpha1/cluster.go

```diff
@@ -566,11 +566,24 @@ func validateNetworking(clusterConfig *Cluster) error {
 	if err != nil {
 		return fmt.Errorf("invalid CIDR block format for Pods: %s. Please specify a valid CIDR block for pod subnet", clusterNetwork.Pods)
 	}
-	_, _, err = net.ParseCIDR(clusterNetwork.Services.CidrBlocks[0])
+	_, serviceCIDRIpNet, err := net.ParseCIDR(clusterNetwork.Services.CidrBlocks[0])
 	if err != nil {
 		return fmt.Errorf("invalid CIDR block for Services: %s. Please specify a valid CIDR block for service subnet", clusterNetwork.Services)
 	}
 
+	if clusterConfig.Spec.DatacenterRef.Kind == SnowDatacenterKind {
+		controlPlaneEndpoint := net.ParseIP(clusterConfig.Spec.ControlPlaneConfiguration.Endpoint.Host)
+		if controlPlaneEndpoint == nil {
```

Comment:
can we move the `validateControlPlaneEndpoint` to this method `validateNetworking` so that we do not need to duplicate cp validation and we wont run into `Endpoint` nil pointer